### DISCUSSION TITLE
Changing how the mention system works

### DISCUSSION THREAD

**Discussion by msmcculley** [ppy/osu#22605] [Ideas]
based on what was changed in #22580, I feel that the ability to mention one's self is not really needed. 

the reasoning behind this is because there should be no reason to mention yourself. It doesnt link back to something you said in chat and definitely doesn't do anything in terms like what discord mentioning does. The systems are just too different to be able to mention yourself. 

Joehuu gave some reasoning as to why they implemented it with being able to mention the user:

- Discord does this (except on dms)
- To set a note for yourself

to me that doesnt necessarily do anything because in discord mentioning oneself tends to have a message attached to it or can be used for a quick redirect for other people who missed the message, the current implementation of the mention doesn't do either. 

In my opinion this is no different than stable's "tab-to-fill" username approach to quickly mentioning... and quite frankly the tab-to-fill method is superior since its 1 button versus needing to right-click then hit mention again, but thats out of scope for what this discussion is for.

**Comment by bdach** [comment]
I don't mind being able to mention self. The mention system is inherently text based. You aren't stopping anything by disallowing it, just making it less accessible.

Autocompletion for mentions would definitely be nice but as you say that's a separate discussion.
